toolbox/process/functions/process_fooof.m

```
@@ -303,7 +307,11 @@
                 params = fmincon(@err_fm_constr,guess,[],[],[],[], ...
                     lb,ub,[],options,fs,spec(chan,:),opt.aperiodic_mode,opt.peak_type);
             catch
-                a = 0; % for catching errors
+                % for catching errors
+            end
+            if isempty(params)
+                % TODO: Improve handling of errors at estimating 'params'
```

@lucwilson, this error handling needs to be improved

---

So I think that we should throw an error if line 307 doesn't run correctly, in which case we either always have 2 +3n params or we error out. So we can integrate a more informative error message for failure to converge in line 307 but otherwise I would remove 312-315.

---

@lucwilson, sounds good to me.

A good approach would be to add an `errMsg` output into `FOOOF_matlab_nll` (as well as in `FOOOF_matlab` and `FOOOF_python`), fill it an early return in the `catch` section. 

So if `errMsg` is not empty in `process_fooof('Run')` the error message will be displayed and added to the report.

---

Addressed in [6cead9f](https://github.com/brainstorm-tools/brainstorm3/pull/729/commits/6cead9f18dd7f45f7b7eebaf1e5041f856c2b185)